job/service.go

```diff
@@ -738,3 +705,120 @@ func populateDownstreamDAGs(dagTree *tree.MultiRootTree, jobSpec models.JobSpec,
 
 	return rootNode, nil
 }
+
+// Refresh fetches all the requested jobs, resolves its dependencies, assign proper priority weights,
+// compile all jobs in the project and upload them to the destination store.
+func (srv *Service) Refresh(ctx context.Context, projectName string, namespaceJobNamePairs []models.NamespaceJobNamePair,
+	progressObserver progress.Observer) (err error) {
+	projectSpec, err := srv.projectService.Get(ctx, projectName)
+	if err != nil {
+		return err
+	}
+
+	// get job specs as requested
+	jobSpecs, err := srv.fetchJobSpecs(ctx, projectSpec, namespaceJobNamePairs, progressObserver)
+	if err != nil {
+		return err
+	}
+
+	// resolve dependency and persist
+	if err := srv.resolveDependency(ctx, projectSpec, jobSpecs, progressObserver); err != nil {
+		// if err is caused by other than asset compilation, ignore the error.
+		if err != nil {
+			if errors.Is(err, errAssetCompilation) {
+				return err
+			}
+		}
+	}
+
+	return srv.deployer.Deploy(ctx, projectSpec, progressObserver)
+}
+
+func (srv *Service) fetchJobSpecs(ctx context.Context, projectSpec models.ProjectSpec,
```

Comment:
fetchJobSpecs should be split into 4 functions, fetchAllForAProject, fetchAllForGivenNamespaces, fetchSpecsForGivenJobNames,  fetchJobSpecs will be the wrapper function